bam_sort.c

```diff
@@ -1686,7 +1686,7 @@ static inline int heap_add_read(heap1_t *heap, int nfiles, samFile **fp,
         if (in_mem[i - nfiles].from < in_mem[i - nfiles].to) {
             size_t from = in_mem[i - nfiles].from;
             heap->entry.bam_record = buf[from].bam_record;
-            if (g_sam_order == TemplateCoordinate) heap->entry.u.key = template_coordinate_keys_get(keys, from);
+            if (g_sam_order == TemplateCoordinate) heap->entry.u.key = buf[from].u.key;
```

Comment:
The bug was restoring the template coordinate keys from the list using the `from` index, but since the buffer was resorted in `sort_blocks`, and the keys in `keys` not updated, the key at the `from` index does match up with the current record.  The fix is to use they reference to the key we already have in `buf[from].u.key`.